assets/js/theme/common/product-details.js

```diff
@@ -56,6 +56,25 @@ export default class ProductDetails {
         this.previewModal = modalFactory('#previewModal')[0];
     }
 
+    /**
+     * https://stackoverflow.com/questions/49672992/ajax-request-fails-when-sending-formdata-including-empty-file-input-in-safari
+     * Safari browser has an issue uploading empty file parameters. This function removes any empty files from the form params
+     * @param form: FormData object
+     * @returns FormData object
+     */
+    filterEmptyFilesFromForm(form) {
+        for (const pair of form.entries()) {
+            const key = pair[0];
+            const val = pair[1];
+            if (typeof val === 'object' && val instanceof File) {
```

Comment:
🍹 Are there any instances where `val instanceof File` passes but `typeof val === 'object'` does not? You can simplify this to just an `instanceof` check if not.

Reply:
guess not - was trying to be safe for the undef case but realized that "undefined instanceof File" is safe

Reply:
would it be better actually to leave the object check and just check if the size/name fields are propertyOf -> and then check the values are not empty/undefined?